src/main/groovy/org/jenkinsci/gradle/plugins/jpi/JpiPlugin.groovy

```diff
@@ -227,28 +227,17 @@ class JpiPlugin implements Plugin<Project> {
     }
 
     private static configureManifest(Project project) {
-        JavaPluginConvention javaPluginConvention = project.convention.getPlugin(JavaPluginConvention)
         TaskProvider<War> jpiProvider = project.tasks.named(JPI_TASK_NAME) as TaskProvider<War>
         TaskProvider<Jar> jarProvider = project.tasks.named(JavaPlugin.JAR_TASK_NAME) as TaskProvider<Jar>
 
-        def configureManifest = project.tasks.register('configureManifest') {
-            it.doLast {
-                Map<String, ?> attributes = attributesToMap(new JpiManifest(project).mainAttributes)
-                jpiProvider.configure {
-                    it.manifest.attributes(attributes)
-                    it.inputs.property('manifest', attributes)
-                }
-                jarProvider.configure {
-                    it.manifest.attributes(attributes)
-                    it.inputs.property('manifest', attributes)
+        [jpiProvider, jarProvider].each { provider ->
+            provider.configure { Jar task ->
+                task.doFirst {
```

Comment:
I think that is too late. The manifest is generated as part of input snapshotting before running this `doFirst`, so the attributes would not make it. I am surprised no test fails.

Reply:
I've been using this method for associating metadata (git and other internal details) with various archival tasks (Jar, Docker, etc).

One thing that the previous version did was to also add the manifest as an input property. I don't do that anymore. The only reason that I could think of doing that was to ensure the task was executed when the class changed. That would anyway happen as the `classes` task would go out of date. Was there any other reason for doing it?